src/ui/App.tsx

```diff
@@ -118,43 +139,55 @@ function App(): JSX.Element {
     },
   };
 
-  useEffect(() => {
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    _signIn();
-  }, []);
-
-  const viewStyle = {flex: 1};
-
-  const cards = [];
-  const threads = threadListState.threads;
   // TODO: Once we take message fetching out of Card creation, only prerender
   // one card below the most recently swiped card. Until then, render more cards
   // and prevent rendering their messages to avoid getting stalled on slow
   // network.
-  let numCardsRendered = 10;
-  while (numCardsRendered-- && numCardsRendered < threads.length) {
-    const threadId = threads[numCardsRendered].id as string;
-    cards.push(
-      <Card
-        key={threadId}
-        threadId={threadId}
-        actions={threadActions}
-        cardSwipedAway={updateThreadListState}
-        preventRenderMessages={numCardsRendered > 2}
-      />,
-    );
-  }
+  const numCardsRendered = 10;
+  const cards = threadListState.threads
+    .slice(0, numCardsRendered)
+    .map((x, index) => {
+      const threadId = defined(x.id);
```

Comment:
Whaaaat is this?

Reply:
Take the first 10 threads and convert them to an array of <Card>s. Use the index to make it so that we only render messages for the first two cards.

^^^ Did that answer your question?